### ISSUE TITLE
False positive for Async assertions and ternary operator

### ISSUE THREAD

**Issue by HunterKohler** [jest-community/eslint-plugin-jest#832]
Eslint is giving me this error:
```
Async assertions must be awaited or returned. (jest/valid-expect)
```
upon the following code (just the second clause of ternary operators):

![carbon (2)](https://user-images.githubusercontent.com/19376599/116013465-4ef6b780-a5fe-11eb-9da0-630defbed600.png)

The script is included in my config's `setupAfterEnv` list.

The error is obviously either
1. I'm an idiot
2. Those are being returned and its a bug

The extensions are just a shortcut macro to have `rejects` and `resolves` for any value. As you can see, this can already done with `expect(...).resolves.not.toThrow()`, and the converse.

**Comment by G-Rath** [comment]
minor nit: when creating issues could you please provide code as snippets instead of as a screenshot? it makes it a lot easier to work with 😅

**Comment by HunterKohler** [comment]
Yes, sorry, I wasn't thinking. Thank you.

**Comment by G-Rath** [comment]
No problem - I'm curious to what program you used to make the screenshot, as it looks perfectly taken.

**Comment by HunterKohler** [comment]
s
